### DISCUSSION TITLE
How to program - after - burning bootloader?

### DISCUSSION THREAD

**Discussion by nlCortana** [SpenceKonde/megaTinyCore#438] [Q&A] [answered]
Hi there,

I'm trying to program an ATtiny1614 using serial port like an arduino UNO/NANO.

I have an breadboard setup with an arduino UNO configured as jtag2updi. I have two different USB->serial adapters. One FTDI based, the other is of a CH340 type. The tiny1614 is soldered on an adapterboard and placed in the breadboard. I can succesfully upload a test-sketch to the ATtiny1614 which outputs "Hello world!" at 9600baud on the serial monitor.
After this :
1. I successfully upload bootloader using  'burn bootloader' (I have show verbose output on uploading turned 'on');
2. I switch to 'Attiny1614 w/Optiboot' in the board menu;
3. Change to the correct serial port;
4. Start compiling / uploading the normal route which fails with the following familiar message:
`         Using Port                           : COM8
         Using Programmer              : arduino
         Overriding Baud Rate          : 115200
avrdude: stk500_recv(): programmer is not responding`

I am aware that the bootloader is only active at the first 8 seconds after startup. I fully remove the Attiny1614 from breadboard and re-seat after arduino software starts programming.

What am I missing?

**Comment by SpenceKonde** [comment, marked as answer]
Select the Optiboot definition when you burn the bootloader. Otherwise you set the fuses and remove any bootloader currently installed. 

Burn bootloader sets ALL fuses, even the ones that say burn bootloader required (these are the ones that, if set wrong, could potentially softbrick it, the equivalent of disabling reset on classic parts), and even the ones that the core does not have anything to do with, in order to initialize the chip to a known state. It then also writes the bootloader if you have a bootloader-using board definition selected.

(normal UIPDI uploads will set the "harmless" fuses with every upload, allowing UPDI sketch upload to write over bootloader instead of uploading and veriftying successfully, (hence trashing the bootloader) but not running either because the fuses put the interrupt vectors 512 bytes  away from where they are - (a failure mode that looks incredibly bizarre and perplexing), and applies the base clock speed setting, and I think something useless but safe, maybe startup time?)_

**Reply by nlCortana** [reply]
Thank you for this reply, you have made my weekend! Now works like a charm. You are amazing.
To me it was a bit counter intuitive to select optiboot board option when general programming via bootloader is not yet active/enabled. Maybe you can add an extra 'how to get started'-explanation in the optiboot section of the readme on the main github page?

**Comment by apws** [comment]
Hi, I am trying latest arduino with latest megaTinyCore with ATtiny1626 and ATtiny3226, both are UPDI-programmable well with schottky and 470ohm (cytron CH340 adapter), even after initial several minutes of reverse polarity and probably smoke on both )), but when I try to select Optiboot version of board and burn the bootloader with (or without) support of DTR autoreset circuit (all placed on breadboard below), the default blink sketch (tested with UPDI) is not flashed it seems. I also initially had problem with COM1 port forced by avrdude.conf, uploading sketch doesnt take selected boarrd COM port, so I patched the global default in conf to my COM10 and then something started to happen - after few tests with 8sec optiboot without DTR (and no success), I tried also the autoreset, now I dont need to power toggle the board but detailed output of uploading lists this only:

avrdude: Version 6.3-20201216
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "C:\Users\petra\AppData\Local\Arduino15\packages\megaTinyCore\hardware\megaavr\2.5.4/avrdude.conf"

         Using Port                    : com10
         Using Programmer              : arduino
         AVR Part                      : ATtiny1626
         Chip Erase delay              : 0 us
         PAGEL                         : P00
         BS2                           : P00
         RESET disposition             : dedicated
         RETRY pulse                   : SCK
         serial program mode           : yes
         parallel program mode         : yes
         Timeout                       : 0
         StabDelay                     : 0
         CmdexeDelay                   : 0
         SyncLoops                     : 0
         ByteDelay                     : 0
         PollIndex                     : 0
         PollValue                     : 0x00
         Memory Detail                 :

                                  Block Poll               Page                       Polled
           Memory Type Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
           ----------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
           signature      0     0     0    0 no          3    0      0     0     0 0x00 0x00
           prodsig        0     0     0    0 no         61   61      0     0     0 0x00 0x00
           fuses          0     0     0    0 no          9   10      0     0     0 0x00 0x00
           fuse0          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse1          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse2          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse4          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse5          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse6          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse7          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           fuse8          0     0     0    0 no          1    0      0     0     0 0x00 0x00
           lock           0     0     0    0 no          1    0      0     0     0 0x00 0x00
           data           0     0     0    0 no          0    0      0     0     0 0x00 0x00
           usersig        0     0     0    0 no         32   32      0     0     0 0x00 0x00
           flash          0     0     0    0 no      16384   64      0     0     0 0x00 0x00
           eeprom         0     0     0    0 no        256   32      0     0     0 0x00 0x00
![unnamed](https://user-images.githubusercontent.com/10888697/147431613-00166cd4-720d-422a-9b8c-7d8100a00306.jpg)

         Programmer Type : Arduino
         Description     : Arduino
         Hardware Version: 3
         Firmware Version: 9.1
         Vtarget         : 0.3 V
         Varef           : 0.3 V
         Oscillator      : 28.800 kHz
         SCK period      : 3.3 us

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.00s

avrdude: Device signature = 0x1e9429 (probably t1626)

avrdude done.  Thank you.

**Reply by SpenceKonde** [reply]
This issue appears unrelated to OP's questions, created new issue to cover this.
